src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -420,6 +440,36 @@ public async Task<uint256> Rewind()
             }
         }
 
+        private void RestoreOutputs(RewindData rewindData)
+        {
+            foreach (UnspentOutputs unspentToRestore in rewindData.OutputsToRestore)
+            {
+                this.logger.LogTrace("Outputs of transaction ID '{0}' will be restored.", unspentToRestore.TransactionId);
+                if (this.unspents.TryGetValue(unspentToRestore.TransactionId, out CacheItem existing))
+                {
+                    existing.UnspentOutputs = unspentToRestore;
+                }
+                else
+                {
+                    this.logger.LogTrace(
+                        "Outputs of transaction ID '{0}' not found in cache, inserting them.",
+                        unspentToRestore.TransactionId);
+                    existing = new CacheItem { UnspentOutputs = unspentToRestore };
+                    this.unspents.Add(unspentToRestore.TransactionId, existing);
+                }
+            }
+        }
+
+        private void RemoveTransactions(RewindData rewindData)
+        {
+            foreach (uint256 transactionToRemove in rewindData.TransactionsToRemove)
+            {
+                this.logger.LogTrace("Attempt to remove transaction with ID '{0}'.", transactionToRemove);
+                if (!this.unspents.ContainsKey(transactionToRemove)) continue;
```

Comment:
Empty line after continue please

Reply:
And also – can it even happen? Is it possible that the transaction that should be removed would not be found?

Reply:
In any case, the remove method will not throw if the element does not exist, so you don't need this line. The only question is whether we should check if remove returns false and if that is an error condition or not